### PR TITLE
Enable swizzling SMEM for transposed dot operand

### PR DESCRIPTION
**Pull request by htyu**
Transposed operand will be accessed in an opposite order from the original operand. Enabling swizzling seems to help performance. I'm seeing 10% performance improvement for our internal model.

### CODE REVIEW

File: include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td [hunk 1]
```diff
@@ -131,6 +131,8 @@ compared to 1*64 when the hasLeadingOffset is false.
 
         if (mfmaEnc) {
           int kDimNum = dotOpEnc.getOpIdx() == 0 ? 1 : 0;
+          if (needTrans)
```
[the remark this answers is not in the document]
Comment (htyu) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: Yes, it's for tt.trans like below. When creating a shared encoding, it looks ahead to see if there's a transpose prior to the dot.

`qk = tl.dot(q, tl.trans(k), allow_tf32=ALLOW_TF32) * alpha`

[the remark this answers is not in the document]
Comment (htyu) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: Yes, this change speeds up our model that has above trans and dot operation by 10%.

[the remark this answers is not in the document]
Comment (htyu) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: BTW, `tl.trans` is used quite often in attention kernel of a transformer mode, iiuc.

[the remark this answers is not in the document]
Comment (zhanglx13) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: cc+ @vgokhale @scxiao

[the remark this answers is not in the document]
Comment (scxiao) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: With this change, do we need a change related to lines at: https://github.com/htyu/triton/blob/ab3aafb4a9158a60a4c3085a13b605f70488d6ff/include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td#L166-L167
We have scenario of non-square tile now.

[the remark this answers is not in the document]
Comment (scxiao) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td:134]: Never mind, this is related to mfma layout

**Pull request merged**